### PR TITLE
[no-typo] rule enhancement checking for a correct prop type spelling

### LINKED ISSUE #213: Rule proposal: validate PropTypes aren't typos
Rule would check that calls to React.PropTypes.whatever are valid. For instance, I have a bad habit of typing `disabled: PropTypes.bools.isRequired` instead of `disabled: PropTypes.bool.isRequired`. I've also done `disabled: propTypes.bool.isRequired` and `disabled: propTypes.bool.isRequired()` by accident.

### PR DESCRIPTION
**Pull request by DianaSuvorova**
closes #213

### CODE REVIEW

File: lib/rules/no-typos.js [hunk 1]
```diff
@@ -21,18 +21,87 @@ const LIFECYCLE_METHODS = [
   'render'
 ];
 
+// https://github.com/facebook/prop-types/blob/master/factoryWithTypeCheckers.js#L96-L114
+const PROP_TYPES = [
```
Comment (ljharb) [inline lib/rules/no-typos.js]: `Object.keys(require('prop-types'))`?

File: lib/rules/no-typos.js [hunk 2]
```diff
@@ -21,18 +21,87 @@ const LIFECYCLE_METHODS = [
...
+
+    function checkValidPropObject (node) {
+      if (node.type === 'ObjectExpression') {
+        node.properties.forEach(prop => checkValidProp(prop.value));
+      }
+    }
+    /* eslint-enable no-use-before-define */
+
     function reportErrorIfClassPropertyCasingTypo(node, propertyName) {
+      if (propertyName === 'propTypes') {
```
Comment (jseminck) [inline lib/rules/no-typos.js]: Should we also perform this check `contextTypes` and `childContextTypes` ?

Comment (DianaSuvorova) [inline lib/rules/no-typos.js]: Good point. Thank for taking a look at this!

File: lib/rules/no-typos.js (hunk 1 above)
Comment (DianaSuvorova) [inline lib/rules/no-typos.js]: Good idea !

**Comment by DianaSuvorova** [comment]
`const { bools } = PropTypes` are not supported. if it is a common use case I would prefer doing it  in a separate PR

**Comment by ljharb** [comment]
It's pretty common (also `import { bool } from 'prop-types'`), but a separate PR is fine.

**Comment by DianaSuvorova** [comment]
@ljharb this should be ready then.  Except for `import { bool } from 'prop-types'` && `const { bools } = PropTypes ` which will be tackled separately

**Pull request merged**